graphql_compiler/schema_transformation/rename_schema.py

```diff
@@ -107,26 +117,148 @@ def rename_schema(
+                        f"{union_name}'s list of union types was None."
+                    )
+                error_message_components.append(
+                    f"Union type {union_type} has no non-suppressed members: "
+                )
+                error_message_components += [
+                    union_member.name.value for union_member in union_type.types
+                ]
+            error_message_components.append(
+                "To fix this, you can suppress the union as well by adding `union_type: None` to "
+                "the `renamings` argument of `rename_schema`, for each value of `union_type` "
+                "described here. Note that adding suppressions may lead to other types, fields, "
+                "unions, etc. requiring suppression so you may need to iterate on this before "
+                "getting a legal schema."
+            )
+        raise CascadingSuppressionError("\n".join(error_message_components))
+
+
+def _ensure_no_unsupported_suppression(
+    schema_ast: DocumentNode, renamings: Mapping[str, Optional[str]], query_type: str
+) -> None:
+    """Confirm renamings contains no enums, interfaces, or interface implementation suppressions."""
+    visitor = SuppressionNotImplementedVisitor(renamings, query_type)
+    visit(schema_ast, visitor)
+    if (
+        not visitor.unsupported_enum_suppressions
+        and not visitor.unsupported_interface_suppressions
+        and not visitor.unsupported_interface_implementation_suppressions
+    ):
+        return
+    # Otherwise, attempted to suppress something we shouldn't suppress.
+    error_message_components = [
+        f"Type renamings {renamings} attempted to suppress parts of the schema for which "
+        f"suppression is not implemented yet."
+    ]
+    if visitor.unsupported_enum_suppressions:
+        error_message_components.append(
+            f"Type renamings mapped these schema enums to None: "
+            f"{visitor.unsupported_enum_suppressions}, attempting to suppress them. However, "
+            f"schema renaming has not implemented enum suppression yet."
+        )
+    if visitor.unsupported_interface_suppressions:
+        error_message_components.append(
+            f"Type renamings mapped these schema interfaces to None: "
+            f"{visitor.unsupported_interface_suppressions}, attempting to suppress them. However, "
+            f"schema renaming has not implemented interface suppression yet."
+        )
+    if visitor.unsupported_interface_implementation_suppressions:
+        error_message_components.append(
+            f"Type renamings mapped these object types to None: "
+            f"{visitor.unsupported_interface_implementation_suppressions}, attempting to suppress "
+            f"them. Normally, this would be fine. However, these types each implement at least one "
+            f"interface and schema renaming has not implemented this particular suppression yet."
```

Comment:
I'm confused about this error message. Am I reading it right, that types that implement any interface cannot be renamed? This is regardless of how the interface is used?

Consider writing a module-level docstring, perhaps, that explains (ideally with some small examples) which operations are and are not supported? I know we can't yet support renaming types that implement an interface when that interface is the exact type of a field, but I was surprised at this.

Reply:
For this PR, types that implement any interface cannot be suppressed (but [they can be renamed](https://github.com/kensho-technologies/graphql-compiler/blob/main/graphql_compiler/tests/schema_transformation_tests/test_rename_schema.py#L171)). I wanted to limit this PR's scope because of the tricky edge cases with interface renaming and there's already a good amount of code here. In a future PR I plan to allow this in some form.

Will document which operations are permitted!